Do not recompile Kotlin buildscripts for non-ABI changes in classpath (#14735)
gradle/gradle: subprojects/docs/src/docs/release/notes.md

While the impact on your build may vary, most builds can expect a noticeably shorter feedback loop when editing Kotlin DSL build logic thanks to this improvement.

> ### Compile avoidance for Kotlin DSL build scripts

Reviewer: 💄 
```suggestion
### Compilation avoidance for Kotlin DSL scripts
```

Until now, any changes to build logic in [buildSrc](userguide/organizing_gradle_projects.html#sec:build_sources) required all the build scripts in the project to be recompiled.

> This release introduces compile avoidance for [Gradle Kotlin DSL](userguide/kotlin_dsl.html) build scripts. 

Reviewer: 💄 
```suggestion
This release introduces compilation avoidance for [Gradle Kotlin DSL](userguide/kotlin_dsl.html) scripts. 
```

> This feature will only cause the build scripts to be recompiled if a change to shared build logic changes the 

Reviewer: 💄 
```suggestion
This feature will cause the scripts to be recompiled only if a change to shared build logic impacts the 
```

ABI (application binary interface) of the resulting artifact.
In simpler terms, changes to private implementation details of build logic, such as private methods or classes, 
bodies of non-private methods or classes,
as well as internal changes to [precompiled script plugins](userguide/custom_plugins.html#sec:precompiled_plugins) 

> such as configuration of tasks will no longer trigger recompilation of the project's build scripts.

Reviewer: ⭕ And that is awesome! :)

> While changes to `buildSrc` immediately affect the classpath of all the buildscripts, this improvement is more general and applies to changes in

Reviewer: 💄 
```suggestion
While changes to `buildSrc` immediately affect the classpath of all the scripts, this improvement is more general and applies to changes in
```

> any jar on the buildscript's classpath that can be added by a plugin applied from an included build or added directly via `buildscript {}` block.

Reviewer: 💄 
```suggestion
any jar on the scripts classpath that can be added by a plugin applied from an included build or added directly via `buildscript {}` block.
```
